src/explorer/DeploymentSlotsNode.ts

```diff
@@ -60,4 +61,60 @@ export class DeploymentSlotsNode extends NodeBase {
     private get azureAccount(): AzureAccountWrapper {
         return this.getTreeDataProvider<AppServiceDataProvider>().azureAccount;
     }
-}
\ No newline at end of file
+
+    public async createNewDeploymentSlot(): Promise<void> {
+        let slotName = '';
+        let slotNameOkay = false;
+
+        // a webapp cannot have more than 4 slots (excluding production)
+        const slotNodes = await this.getChildren();
+        const slotLabels = slotNodes.map(node => {
+            return node.label;
+        });
+
+        while (!slotNameOkay) {
+            slotName = await window.showInputBox({
+                prompt: 'Enter a unique name for the new deployment slot',
+                validateInput: (value: string) => {

```

Comment:
Duplicated code, is it possible to share the same code with Web App name validation?